### PR TITLE
Scalable board

### PR DESCRIPTION
**Pull request by fabientownsend**

### CODE REVIEW

File: spec/game_play_spec.rb [hunk 1]
```diff
@@ -163,11 +161,13 @@
   end
 
   it "shoudl ask a second time when value is too low with menus" do
```
Comment (gemcfadyen) [inline spec/game_play_spec.rb:163]: 'shoudl' should be 'should' in testname 😸

File: lib/marks.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 module Mark
   ROUND = "O"
   CROSS = "X"
+  EMPTY = " "
```
Comment (gemcfadyen) [inline lib/marks.rb:4]: Is EMPTY really a Mark? 🤔

File: lib/board.rb [hunk 3]
```diff
@@ -62,11 +79,11 @@ def win_with_diagonal?(mark)
...
-    [0, 1, 2].all? { |index| board[index][index] == mark }
+    board.size.times.all? { |index| board[index][index] == mark }
   end
 
   def check_backward_diagonal?(mark)
-    [0, 1, 2].all? { |index| board[index][backward_index(index)] == mark }
+    board.size.times.all? { |index| board[index][backward_index(index)] == mark }
   end
 
   def backward_index(index)
```
Comment (gemcfadyen) [inline lib/board.rb:89]: I think it would be worth adding a couple of tests to make sure a winning row is found on a 4x4 board.

**Review by gemcfadyen — COMMENTED** [review]
I'm late to the review but a couple of minor points on your PR from last week. 👍

File: lib/board.rb (hunk 3 above)
Comment (fabientownsend) [reply lib/board.rb:89]: Yes, I will had that!

File: spec/game_play_spec.rb (hunk 1 above)
Comment (fabientownsend) [reply spec/game_play_spec.rb:163]: I should edit that :D

File: lib/marks.rb (hunk 2 above)
Comment (fabientownsend) [reply lib/marks.rb:4]: I agree, it seems a bit weird. I removed it as I need the id for the empty spot.

**Pull request closed without merging**